pkg/controller/volume/attachdetach/attach_detach_controller_test.go

```diff
@@ -136,6 +138,87 @@ func Test_AttachDetachControllerStateOfWolrdPopulators_Positive(t *testing.T) {
 	}
 }
 
+func Test_AttachDetachControllerStateOfWorldPopulatorsMigration_Positive(t *testing.T) {
+	// defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, false)()
+	// defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationGCE, false)()
+	// defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIDriverRegistry, false)()
+
+	fakeKubeClient := controllervolumetesting.CreateTestClient()
+	informerFactory := informers.NewSharedInformerFactory(fakeKubeClient, time.Second*1)
+	nodeInformer := informerFactory.Core().V1().Nodes().Informer()
+	csiNodeInformer := informerFactory.Storage().V1().CSINodes().Informer()
+	podInformer := informerFactory.Core().V1().Pods().Informer()
+
+	stopCh := make(chan struct{})
+
+	adc, _, _, _ := testCaseSetup(t, fakeKubeClient, informerFactory, nodeInformer, csiNodeInformer, podInformer, stopCh)
+
+	// Populate ASW
+	err := adc.populateActualStateOfWorld()
+	if err != nil {
+		t.Fatalf("Run failed with error. Expected: <no error> Actual: <%v>", err)
+	}
+
+	// Populate DSW
+	err = adc.populateDesiredStateOfWorld()
+	if err != nil {
+		t.Fatalf("Run failed with error. Expected: <no error> Actual: %v", err)
+	}
+
+	go adc.reconciler.Run(stopCh)
+	go adc.desiredStateOfWorldPopulator.Run(stopCh)
+	defer close(stopCh)
+
+	// Test that volumes in desiredStateOfWorld and actualStateOfWorld are migrated
+	csiNodesList, err := informerFactory.Storage().V1().CSINodes().Lister().List(labels.Everything())
+	for _, node := range csiNodesList {
+		attachedVolumes := adc.actualStateOfWorld.GetAttachedVolumesForNode(types.NodeName(node.Name))
+		for _, volume := range attachedVolumes {
+			if volume.VolumeSpec == nil {
+				continue
+			}
+			if volume.VolumeSpec.Volume != nil {
+				if volume.VolumeSpec.Volume.CSI != nil {
```

Comment:
doesn't `Volume.CSI != nil` imply that this is a CSI volume? Why is the error message that it `failed to translate to CSI`. That seems like the opposite of what this means